attachment_preview/model/ir_attachment.py

```diff
@@ -32,35 +32,47 @@ def get_binary_extension(
             self, cr, uid, model, ids, binary_field, filename_field=None,
             context=None):
         result = {}
-        for this in self.pool[model].browse(
-                cr, uid,
-                ids if isinstance(ids, collections.Iterable) else [ids],
-                context=context):
-            if not this.id:
-                result[this.id] = False
-                continue
-            extension = ''
-            if filename_field and this[filename_field]:
-                filename, extension = os.path.splitext(this[filename_field])
-            if not this[binary_field]:
-                result[this.id] = False
-                continue
-            if not extension:
-                try:
-                    import magic
-                    ms = magic.open(
-                        hasattr(magic, 'MAGIC_MIME_TYPE') and
-                        magic.MAGIC_MIME_TYPE or magic.MAGIC_MIME)
-                    ms.load()
-                    mimetype = ms.buffer(
-                        base64.b64decode(this[binary_field]))
-                except ImportError:
-                    (mimetype, encoding) = mimetypes.guess_type(
-                        'data:;base64,' + this[binary_field], strict=False)
-                extension = mimetypes.guess_extension(
-                    mimetype.split(';')[0], strict=False)
-
-            result[this.id] = (extension or '').lstrip('.').lower()
+        # First pass: load fields in bin_size mode to avoid loading big files
+        #  unnecessarily.
+        if filename_field:
+            context_bin_size = dict(context or {})
+            context_bin_size['bin_size'] = True
+            ids_to_browse = ids if isinstance(ids, collections.Iterable) \
+                else [ids]
+            for this in self.pool[model].browse(cr, uid, ids_to_browse,
+                                                context=context_bin_size):
+                if not this.id:
+                    result[this.id] = False
+                    continue
+                extension = ''
+                if this[filename_field]:
+                    filename, extension = os.path.splitext(
+                        this[filename_field])
+                if this[binary_field]:
+                    result[this.id] = extension
+                else:
+                    result[this.id] = False
+        # Seconds pass for all attachments which have to be loaded fully
+        #  to get the extension from the content
+        ids_to_browse = [_id for _id in ids_to_browse if _id not in result]
```

Comment:
how can any id be left here in the happy flow? That is, don't you effectively disable mime type tests?